external/test-runtime/optional.csproj

```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>net45</TargetFramework>
+    <TargetFrameworkIdentifier>.NETFramework</TargetFrameworkIdentifier>
```

Comment:
This should be happening implicitly by the Microsoft.NET.Sdk:

[https://github.com/dotnet/sdk/blob/33d51eab4d1e8dccf90eec946cf311ef29e0857f/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L49-L72](https://github.com/dotnet/sdk/blob/33d51eab4d1e8dccf90eec946cf311ef29e0857f/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L49-L72)

I don't think this is needed.

Reply:
If I don't set it, it picks the default value `.NETPortable`, which is wrong. I believe the properties are evaluated before the TargetFramework is set here because the target is imported with the SDK before the `TargetFramework` property is set.